File: Valid Parentheses/valid_parantheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
 var isValid = function(s) {
    const BRACKETS_MAPPING = { '{': '}', '[':']','(':')' }
    s = s.split('')
    let open = []
    while(s.length) {
        if(s[0] in BRACKETS_MAPPING)
            open.push(BRACKETS_MAPPING[s[0]])
        else if(s[0] !== open.pop())
            return false
        s.shift()
    }
    return !open.length
};

console.log(isValid("()"))
console.log(isValid("()[]{}"))
console.log(isValid("(]"))
console.log(isValid("{[]}"))
console.log(isValid("(])"))

File: Valid Parentheses/valid_parantheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        BRACKETS_MAPPING = { '{': '}', '[':']','(':')' }
        open = list()
        for bracket in s:
            if bracket in BRACKETS_MAPPING:
                open.append(BRACKETS_MAPPING[bracket])
            elif not(len(open)) or bracket != open.pop():
                return False
        return not(len(open))

            



classInstance = Solution()
print(classInstance.isValid("()"))
print(classInstance.isValid("()[]{}"))
print(classInstance.isValid("(]"))
print(classInstance.isValid("{[]}"))
print(classInstance.isValid("(])"))
print(classInstance.isValid("]"))